src/main/java/net/openhft/chronicle/bytes/MappedFile.java

```diff
@@ -46,7 +47,8 @@
 public abstract class MappedFile extends AbstractCloseableReferenceCounted {
     static final boolean RETAIN = Jvm.getBoolean("mappedFile.retain");
     private static final long DEFAULT_CAPACITY = 128L << 40;
-    protected final String canonicalPath;
+    // The canonical path is pre-pended with static and random data to prevent unrelated synchronization on internalized Strings
+    protected final String internalizedToken;
```

Comment:
I understand why you changed this, - so, If you think this should be private, please make it private and provided access to depending class via a method call, instead of having a protected field.